### ISSUE TITLE
Load pretrained checkpoints when tuning?

### ISSUE THREAD

**Issue by zhenduow** [facebookresearch/DPR#44]
Hi,

I searched and did't find this issue. (but maybe I am wrong) 

I get  'Checkpoint files []'  message when tuning on my own dataset. So I wonder if this means that I tune the original bert-uncased-base model on my dataset.

Is there an option for loading checkpoint that's trained on NQ when tuning? I see there is an option `pretrained_file`. But no matter what I pass to it, I still get the 'Checkpoint files []' message...

Thank you!

**Comment by vlad-karpukhin** [comment]
Hi,
--pretrained_file parameter is only for fairseq or pytext based model's initialization.

"Checkpoint files ..." log message only appears when you don't specify --model_file parameter.
The logic is as follows:
- it first checks if  --model_file is specified. If yes, just use that to load the bi-encoder model weights.
- otherwise it tries to find the latest saved checkpoint file in the output (specified by --output_dir) folder and use that instead. This is done to resume training automatically after the training process crashes or stops inside some cluster job management tools like SLURM.
- if no --model_file or checkpoint files are found in --output_dir, the code just uses 'raw' model initialized from either BERT or RoBerta pretrained models.

So, if you want to fine-tune over our trained model, you will need to specify it by --model_file argument.
But, there may be some issue with our provided trained models for finetuning - we also save gradients momentum, learning rate schedule state, epoch and an iteration number inside that epoch when the model checkpoint was taken and then initialize training state from them. I might have deleted those extra data from the shared checkpoints, but let me know if you will see some issues with finetuning.

Also, overall recommendation is not to finetune for specific dataset only but train it jointly with NQ&Trivia.

**Comment by zhenduow** [comment]
Hi vlad-karpukhin,

I use `--pretrained_file bert-base-encoder.cp` and `--output_dir checkpoint/retriever/multiset`. But it still shows me checkpoint files []...

Where is this wrong? Thank you!

**Comment by vlad-karpukhin** [comment]
Again, --pretrained_file is just to initialize pytext/fairseq models?
Do you use any of those (i.e what is your --encoder_type)?

The output dir is search for specific file pattern (defined by --checkpoint_file_name which is dpr_biencoder by default).

**Comment by zhenduow** [comment]
This is my full argument list:
`python train_dense_encoder.py --encoder_model_type hf_bert --model_file bert-base-encoder.cp --pretrained_model_cfg bert-base-uncased --train_file ../data/DPR_question_retriever_training --dev_file ../data/DPR_question_retriever_training --output_dir checkpoint/retriever/multiset`

**Comment by vlad-karpukhin** [comment]
Since you are using --encoder_model_type hf_bert (i.e.  HuggingFace model code, not fairseq or pytext), you don't need to specify --pretrained_file . It will basically be ignored.

If it is able to find your bert-base-encoder.cp, you should see the messages like 
"Reading saved model from ..." 
You specify bert-base-encoder.cp without any path, is it located in the process's current dir.
Looks like the code is not able to find it there.

**Comment by zhenduow** [comment]
I changed the path to full. Now it works!

I thought the path would be appended to the `output_dir`.

Thank you very much!
